### DISCUSSION TITLE
Installer doesn't work if Sunshine is running (Windows)

### DISCUSSION THREAD

**Discussion by c---** [LizardByte/community#480] [Sunshine]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your issue described in the documentation?

- [X] I have read the documentation

### Is your issue present in the latest beta/pre-release?

This issue is present in the latest pre-release

### Describe the Bug

In Windows currently it's not possible to upgrade Sunshine from within a Sunshine session.
```
Error opening file for writing:
C:\Program Files\Sunshine\sunshine.exe
```
Which is understandable since it's running. However, Windows has a mechanism to replace files during reboot. The Sunshine installer should use this so that the installation can complete while running in a Sunshine session. Otherwise Sunshine must be turned off and then there may be no easy way to access the machine to perform the upgrade. This happens for example when Sunshine is running in a virtual machine.

### Expected Behavior

Installer should complete and request a reboot.

### Host Operating System

Windows

### Operating System Version

11

### Architecture

64 bit

### Sunshine commit or version

v2024.707.141913 Pre-release

### Package

Windows - installer

### GPU Type

n/a

### GPU Model

N/A

### GPU Driver/Mesa Version

N/A

### Capture Method

None

### Config

```Shell
N/A
```

### Relevant log output

```shell
N/A
```

**Comment by reloxx13** [comment]
If you choose uninstall previous version it will shutdown. beware, someone mentioned his config was gone by doing this.
i cannot confirm this, i always hit uninstall and the config got never deleted.
